R/utils.R

```diff
@@ -60,6 +60,46 @@ find_phantom <- function() {
   path.expand(path)
 }
 
+phantomjs_cmd_result = function(args, wait = TRUE) {
+  # Retrieve and store output from STDOUT
+  utils::capture.output(invisible(phantom_run(args = args, wait = wait)),
```

Comment:
Is `invisible` needed here?

Reply:
Sort of.

Using `invisible()` switches the output when `phantomjs` is not installed from `"NULL"` (character `NULL`) to displaying an empty string (e.g. `character(0)`), which I think is a more graceful failure.

Reply:
Given that this function returns the value of `capture.output` and doesn't print the return value of `phantom_run`, I don't think that the `invisible` makes a difference here. If that's wrong, it would help me understand if you would provide the output with and without `invisible()` present.

Reply:
## PhantomJS not found: 
 
```r
# Not installed phantomjs
# invisible missing
R> webshot:::phantomjs_version()
[1] "NULL"
```

```r
# Not installed phantomjs
# invisible present
R> webshot:::phantomjs_version()
character(0)
```

## Installed versions of phantomjs:

```r
# invisible not present
R> webshot:::phantomjs_version()
[1] "2.1.1" "[1] 0"
```

```r
# invisible present
R> webshot:::phantomjs_version()
[1] "2.1.1"
```

## Conclusion

After checking both options, I'm switching from _sort of_ to **`invisible()` is needed**.

Reply:
In short, when using `capture.output()` we are focusing solely on what is sent to console. Since the function is using `phantomjs_run()`, this means that a status code for the process is included in addition to the output. Thus, we would end up getting more entries than desired as shown above.